src/topicFactory.js

```diff
@@ -0,0 +1,93 @@
+import nets from 'nets'
+import qs from 'qs'
+import randomString from './util/randomString'
+const CHASQUI_URL = 'https://chasqui.uport.me/api/v1/topic/'
+
+function TopicFactory (isOnMobile, pollingInterval = 2000, chasquiUrl = CHASQUI_URL) {
+  function waitForHashChange (topicName, cb) {
```

Comment:
yeah, but saying it doesn't look like the function actually returns the function